string location # North, East, South, West
bool is_passing